app/helpers/menu_helper.rb

```diff
@@ -85,6 +85,19 @@ def forcibly_denied_switch?(switch)
   end
 
   def api_selector_services
-    @api_selector_services ||= site_account.provider? && logged_in? ? current_user.accessible_services.includes(:account) : Service.none
+    @api_selector_services ||= (site_account.provider? && logged_in? ? current_user.accessible_services.includes(:account) : Service.none).decorate
   end
+
+  def audience_link
+    if can?(:manage, :partners)
+      admin_buyers_accounts_path
+    elsif can?(:manage, :finance)
+      admin_finance_root_path
+    elsif can?(:manage, :portal)
+      provider_admin_cms_templates_path
+    elsif can?(:manage, :settings)
+      edit_admin_site_usage_rules_path
```

Comment:
maybe we need an else statement too just in case:

```suggestion
      else
        '#'
```